packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -544,61 +633,55 @@ export function makeVirtualObjectManager(
           },
         });
       }
-      harden(wrappedState);
+      harden(state);
 
       if (initializing) {
         cache.remember(innerSelf);
       }
-      const self = actualize(wrappedState);
       let toHold;
       let toExpose;
-      const facetiousness = assessFacetiousness(self);
-      switch (facetiousness) {
-        case 'one': {
-          toHold = Far(tag, self);
-          vrm.checkOrAcquireFacetNames(kindID, null);
-          toExpose = toHold;
-          break;
-        }
-        case 'many': {
-          toExpose = {};
-          toHold = [];
-          const facetNames = Object.getOwnPropertyNames(self).sort();
-          assert(
-            facetNames.length > 1,
-            'a multi-facet object must have multiple facets',
-          );
-          vrm.checkOrAcquireFacetNames(kindID, facetNames);
-          for (const facetName of facetNames) {
-            const facet = Far(`${tag} ${facetName}`, self[facetName]);
-            toExpose[facetName] = facet;
-            toHold.push(facet);
-            facetToCohort.set(facet, toHold);
-          }
-          harden(toExpose);
-          break;
+      unweakable.add(state);
+      if (facetNames === null) {
+        const context = { state };
+        context.self = bindMethods(context, behaviorTemplate);
+        toHold = Far(tag, context.self);
+        // eslint-disable-next-line no-proto
+        linkToCohort.set(toHold.__proto__, toHold);
```

Comment:
Is there any benefit to using `Object.getPrototypeOf` instead of `__proto__`? What happens if someone defines a method named `__proto__` in their `behavior`?

Reply:
Oh god yes! Please avoid `__proto__`. Not only can `Object.prototype.__proto__` be overridden, Caridy convinced us and tc39 that secure environments should probably remove it on initialization. SES does not yet do so, but probably will and should (attn @kriskowal @mhofman ). tc39 made in normative optional, so implementations are allowed to not provide it in the first place (as a future SES/XS may elect to do).

Reply:
This was a case of my own ignorance coming into play.  I had never before, in years of JS development, ever had occasion to want to get to the prototype of an object.  Fortunately, this is a very easy fix.